### DISCUSSION TITLE
How to approach parameter settings

### DISCUSSION THREAD

**Discussion by IlyaShaternik** [coin-or/Cbc#500] [Q&A] [answered]
Hi! I'm relatively new to MILP solving, so I'd like to apologize if my question is stupid. But I'm hoping someone may give me at least some general advice how to approach my problem.

So I have a MILP model, which I need to solve using CBC. I tested the model using Gurobi, and it manages to find feasible solution with sufficient relative gap (10%) in about 15 minutes. Every new incumbent is found by some heuristic. Down to a gap of 15% all new solutions are found at the root.

I've tried solving the model with CBC with default settings, but have found no success. I've used release 2.10.7. The only feasible solutions found by CBC are found during Feasibility Pump. After that no new solution is found for several hours (I didn't run it longer than 8 hours). I've tried changing some parameters, adding different heuristics, but it didn't help. The only improvement I managed to achieve is changing `when_` of feasibility pump from 13 to 1, which allowed Feasibility Pump to find better solution. 

Without modifying model, what can I try to achieve better performance? 

[model.zip](https://github.com/coin-or/Cbc/files/8660679/model.zip)

**Comment by jjhforrest** [comment]
Will look at this next week.  What value solution did Gurobi get?

The best I could do was 9.7e6 in 12 minutes.

**Reply by IlyaShaternik** [reply]
Hi. Thanks for the reply! Just in case, I've rerun Gurobi and got solution value of 6.79e+06 in about 28 minutes. Usually it takes about 15 minutes, so increase in time is probably caused by machine workload during execution. Also here is Gurobi' log: [gurobi_solution.log](https://github.com/coin-or/Cbc/files/8671696/gurobi_solution.log)

**Comment by jjhforrest** [comment, marked as answer]
I have done a bit more work - but only with partial success.  The best I 
could get the code to do was 7.4e6 at root node in 15 minutes.  This was 
got by setting a "fake" cutoff of 7.5e6 and using multiple feasibility 
pumps.  I did three rounds each on four cpus - by -multiplerootpasses 
412. I think the problem is that there are large numbers of general 
integer variables and the version of the feasibility pump in Cbc does 
not like general integer variables.  There has been a paper on 
feasibility pump published which I may try and use to improve the 
feasibility pump.

The other area which might help is if I can see how to improve the 
Dantzig Wolfe heuristic as the problem does decompose.  However I am not 
at all familiar with that piece of code. - The preprocessed model I was 
working with had the following structure -

32405 rows, 26536 columns and 9 blocks
6545 master rows and 4575 columns outside blocks
block 1 has 3326 rows and 2466 columns
block 2 has 1068 rows and 1020 columns
block 3 has 2288 rows and 2196 columns
block 4 has 2244 rows and 2152 columns
block 5 has 1017 rows and 968 columns
block 6 has 500 rows and 460 columns
block 7 has 9680 rows and 8341 columns
block 8 has 2762 rows and 1932 columns
block 9 has 2975 rows and 2426 columns


John Forrest
